src/NuGet.Core/NuGet.Common/NuGetEnvironment.cs

```diff
@@ -85,22 +76,23 @@ public static string GetFolderPath(SpecialFolder folder)
                                 () => GetEnvironmentVariable("PROGRAMDATA"),
                                 () => GetEnvironmentVariable("ALLUSERSPROFILE"));
                         }
+                        else if (RuntimeEnvironmentHelper.IsMacOSX)
+                        {
+                            return Path.Combine(@"/","Library", "Application Support");
```

Comment:
Since you know the platform and you're building an absolute path, there's no need to use `Path.Combine` here. Just use the literal path `/Library/Application Support`